File: aerolinea-backend/src/main/java/com/halcon/aerolineas/utils/JsonResponse.java
package com.halcon.aerolineas.utils;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import java.time.LocalDate;
import java.time.LocalDateTime;

public class JsonResponse {
    private static final Gson gson = new GsonBuilder()
            .setPrettyPrinting()
            .registerTypeAdapter(LocalDate.class, new LocalDateAdapter())
            .registerTypeAdapter(LocalDateTime.class, new LocalDateTimeAdapter())
            .create();
    
    public static String success(Object data) {
        ResponseWrapper wrapper = new ResponseWrapper(true, data, null);
        return gson.toJson(wrapper);
    }
    
    public static String success(String message, Object data) {
        ResponseWrapper wrapper = new ResponseWrapper(true, data, message);
        return gson.toJson(wrapper);
    }
    
    public static String error(String message) {
        ResponseWrapper wrapper = new ResponseWrapper(false, null, message);
        return gson.toJson(wrapper);
    }
    
    public static String toJson(Object obj) {
        return gson.toJson(obj);
    }
    
    public static <T> T fromJson(String json, Class<T> classOfT) {
        return gson.fromJson(json, classOfT);
    }
    
    private static class ResponseWrapper {
        private boolean success;
        private Object data;
        private String message;
        
        public ResponseWrapper(boolean success, Object data, String message) {
            this.success = success;
            this.data = data;
            this.message = message;
        }
    }
}

File: Bedly1/HotelesAPI/Utils/JsonResponse.cs
namespace HotelesAPI.Utils
{
    public class JsonResponse
    {
        public bool Success { get; set; }
        public string Message { get; set; } = string.Empty;
        public object? Data { get; set; }

        public static JsonResponse Ok(string message, object? data = null)
        {
            return new JsonResponse { Success = true, Message = message, Data = data };
        }

        public static JsonResponse Error(string message)
        {
            return new JsonResponse { Success = false, Message = message, Data = null };
        }
    }
}